orm: cgen error when using shared

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.5.0 75fc8be, press to see full `v doctor` output</summary>

|V full version      |V 0.5.0 75fc8bed1876a3dccbdea5005125473ac9fbf8cb
|:-------------------|:-------------------
|OS                  |macos, macOS, 26.2, 25C56
|Processor           |10 cpus, 64bit, little endian, Apple M4
|Memory              |0.21GB/16GB
|                    |
|V executable        |/Users/leodev/p/v/v/v
|V last modified time|2026-01-31 10:45:14
|                    |
|V home dir          |OK, value: /Users/leodev/p/v/v
|VMODULES            |OK, value: /Users/leodev/.vmodules
|VTMP                |OK, value: /tmp/v_501
|Current working dir |OK, value: /Users/leodev/p/v/test
|                    |
|Git version         |git version 2.52.0
|V git status        |weekly.2026.05-80-g75fc8bed
|.git/config present |true
|                    |
|cc version          |Apple clang version 17.0.0 (clang-1700.0.13.3)
|gcc version         |Apple clang version 17.0.0 (clang-1700.0.13.3)
|clang version       |Apple clang version 17.0.0 (clang-1700.0.13.3)
|tcc version         |tcc version 0.9.28rc 2026-01-10 HEAD@5ec0e6f8 (AArch64 Darwin)
|tcc git status      |thirdparty-macos-arm64 f995efa3
|emcc version        |N/A
|glibc version       |N/A

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  repro.v && repro`
```v
import db.sqlite

fn main() {
	shared x := sqlite.connect(':memory:')!

	lock x {
		sql x {
		}!
	}
}

```

**What did you see?**
```
================== C compilation error (from cc): ==============
cc: /tmp/v_501/repro.01KGAWBBPN3RPDCRYHFHT6C516.tmp.c:12273:83: error: use of undeclared identifier '_orm__Connection___shared__db__sqlite__DB_index'; did you mean '_orm__Connection_db__sqlite__DB_index'?
cc:  12273 |                 orm__Connection _t2 = (orm__Connection){.___shared__db__sqlite__DB = x, ._typ = _orm__Connection___shared__db__sqlite__DB_index};
cc:        |                                                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc:        |                                                                                                 _orm__Connection_db__sqlite__DB_index
cc: /tmp/v_501/repro.01KGAWBBPN3RPDCRYHFHT6C516.tmp.c:3468:11: note: '_orm__Connection_db__sqlite__DB_index' declared here
cc:  3468 | const u32 _orm__Connection_db__sqlite__DB_index = 0;
cc:       |           ^
cc: /tmp/v_501/repro.01KGAWBBPN3RPDCRYHFHT6C516.tmp.c:12273:44: error: field designator '___shared__db__sqlite__DB' does not refer to any field in type 'orm__Connection' (aka 'struct orm__Connection')
cc:  12273 |                 orm__Connection _t2 = (orm__Connection){.___shared__db__sqlite__DB = x, ._typ = _orm__Connection___shared__db__sqlite__DB_index};
cc:        |                                                         ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc: 2 errors generated.
================================================================
Try passing `-g` when compiling, to see a .v file:line information, that correlates more with the C error.
(Alternatively, pass `-show-c-output`, to print the full C error message).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

Not a cgen error

--- a/vlib/v/gen/c/orm.v
+++ b/vlib/v/gen/c/orm.v
@@ -153,7 +153,7 @@ fn (mut g Gen) sql_stmt_line(stmt_line ast.SqlStmtLine, connection_var_name stri
 fn (mut g Gen) write_orm_connection_init(connection_var_name string, db_expr &ast.Expr) {
 	db_expr_type := g.get_db_expr_type(db_expr) or { verror('ORM: unknown db type for ${db_expr}') }
 
-	mut db_ctype_name := g.styp(db_expr_type)
+	mut db_ctype_name := g.styp(db_expr_type.clear_flag(.shared_f))
 	is_pointer := db_ctype_name.ends_with('*')
 	reference_sign := if is_pointer { '' } else { '&' }
 	db_ctype_name = db_ctype_name.trim_right('*')
@@ -166,7 +166,13 @@ fn (mut g Gen) write_orm_connection_init(connection_var_name string, db_expr &as
 		g.writeln(';')
 	} else {
 		g.write('(orm__Connection){._${db_ctype_name} = ${reference_sign}')
+		if db_expr_type.has_flag(.shared_f) {
+			g.write('&')
+		}
 		g.expr(db_expr)
+		if db_expr_type.has_flag(.shared_f) {
+			g.write('->val')
+		}
 		g.writeln(', ._typ = _orm__Connection_${db_ctype_name}_index};')
 	}
 }

